homeassistant/helpers/script.py

```diff
@@ -983,35 +973,56 @@ async def _async_if_step(self) -> None:
             with trace_path("else"):
                 await self._async_run_script(if_data["if_else"])
 
+    def _async_futures_with_timeout(
+        self,
+        timeout: float | None,
+    ) -> tuple[
+        list[asyncio.Future[None]],
+        asyncio.TimerHandle | None,
+        asyncio.Future[None] | None,
+    ]:
+        """Return a list of futures to wait for.
+
+        The list will contain the stop future.
+
+        If timeout is set, a timeout future and handle will be created
+        and will be added to the list of futures.
+        """
+        timeout_handle: asyncio.TimerHandle | None = None
+        timeout_future: asyncio.Future[None] | None = None
+        futures: list[asyncio.Future[None]] = [self._stop]
+        if timeout:
```

Comment:
I didn't consider that someone might specify a zero timeout

https://github.com/home-assistant/core/pull/115830